engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -129,37 +140,89 @@ public NadelExecutionStrategy(List<Service> services,
 
             boolean usesDynamicService = fieldExecutionStepInfo.getFieldDefinition().getDirectives()
                     .stream()
-                    .anyMatch(directive -> directive.getName().equals(NadelDirectives.DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.getName()));
+                    .anyMatch(directive -> directive.getName().equals(DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.getName()));
```

Comment:
Note for self - graphql-java should have `hasDirective("name)`

The above can be simplied via `fieldDef.getDirective(DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.getName()) != null`